src/diff.rs

```diff
@@ -74,8 +74,46 @@ pub struct DiffStats {
     raw: *mut raw::git_diff_stats,
 }
 
+/// Structure describing the binary contents of a diff.
+pub struct DiffBinary<'diff> {
+    raw: *const raw::git_diff_binary,
+    _marker: marker::PhantomData<&'diff raw::git_diff_binary>,
```

Comment:
Currently this is largely being expressed as:

```rust
_marker: marker::PhantomData<&'diff Diff>,
```

which is to say the `PhantomData` types tend to hold the Rust types rather than the C ones